### PR TITLE
Add initial version of Ghost+nginx+MySQL pattern

### PR DESCRIPTION
**Pull request by cjihrig**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    && export CONSUL_TEMPLATE_CHECKSUM=31dda6ebc7bd7712598c6ac0337ce8fd8c533229887bd58e825757af879c5f9f \
+    && curl --retry 7 --fail -Lso /tmp/consul-template.zip "https://releases.hashicorp.com/consul-template/${CONSUL_TEMPLATE_VERSION}/consul-template_${CONSUL_TEMPLATE_VERSION}_linux_amd64.zip" \
+    && echo "${CONSUL_TEMPLATE_CHECKSUM}  /tmp/consul-template.zip" | sha256sum -c \
+    && unzip /tmp/consul-template.zip -d /usr/local/bin \
+    && rm /tmp/consul-template.zip
+
+# Add Containerpilot and set its configuration
+COPY etc/containerpilot.json5 /etc
+ENV CONTAINERPILOT /etc/containerpilot.json5
+ENV CONTAINERPILOT_VERSION 3.3.3
```
Comment (geek) [inline Dockerfile]: might as well use 3.4.2 (https://github.com/joyent/containerpilot/releases/tag/3.4.2)

File: bin/config-ghost.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  throw new Error('run in an unexpected mode');
+}
+
+lookupService((err, service) => {
+  if (err) {
+    throw err;
+  }
+
+  debug('service = %j', service);
+  createGhostConfigFile(service);
```
Comment (geek) [inline bin/config-ghost.js:22]: service may not exist here... if mysql goes down

File: etc/containerpilot.json5 [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  logging: {
+    level: '{{ .LOG_LEVEL | default "INFO" }}'
+  },
+  jobs: [
+    {
+      name: 'preStart',
+      exec: 'node /bin/config-ghost.js preStart',
+      when: {
+        source: 'watch.mysql-primary',
+        each: 'changed'
```
Comment (geek) [inline etc/containerpilot.json5]: maybe change to 
`once: 'healthy'`

File: bin/config-ghost.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+'use strict';
+const Cp = require('child_process');
+const Consulite = require('consulite');
+const consulHost = `${process.env.CONSUL}:8500`;
```
Comment (geek) [inline bin/config-ghost.js]: should use `localhost` if using CONSUL_AGENT

**Review by geek — CHANGES_REQUESTED** [review]

File: bin/config-ghost.js (hunk 2 above)
Comment (cjihrig) [reply bin/config-ghost.js:22]: We just looked up the service a couple lines up. If it wasn't found, an error is thrown. Technically, the service could have gone down between then and now, but I don't think we can handle that.

**Review by geek — APPROVED** [review]

**Pull request merged**